connman/plugins/ofono.c

```diff
@@ -1680,6 +1681,10 @@ static void netreg_update_regdom(struct modem_data *modem,
 	char *alpha2;
 	int mcc;
 
+	/* Do not change regdom here if it is set to follow timezone. */
+	if (connman_setting_get_bool("RegdomFollowsTimezone"))
```

Comment:
Would it make sense to have a method to use SIM if possible, and timezone only as a backup?

Reply:
If timezone is set to automatic and updated via another service, or set via ConnMan even it should be the cellular stack changing it -> the change should be noted when RegdomFollowsTimezone is set. I was just thinking if it can create some corner cases if used only as a backup in situations where Ofono plugin sets the regdom to X and for some reason user has selected a certain timezone, which should be preferred as in that case it is unknown which option the user wishes the system to follow and this gives the power to do so.

Reply:
At least I (as a user) wouldn't expect the choice of timezone to affect the wifi regdom.

Reply:
This is just the case, relying on timezone would be a kind of solution for every possible case. Having cellular support and a SIM present is a somewhat special case ;).

But I think many do not even expect that the cellular stack changes the regdom? 

To facilitate this fully to have timezone as a backup would need quite a lot of new code I think. `technology.c` has no way of detecting who has changed the value. But would you suggest that the timezone at startup defines the base regdom and then that is changed by Ofono plugin? Eventually the timezone after all changes, or should change, according to the cellular network when it is supported -> both produce the same result. 

My argument here is that now with this feature user can keep a timezone and regdom.

Reply:
IMO network registration should take precedence. When available, it's a far more accurate guess than whatever the user has selected in the settings.